src/framework/vst/internal/vstplugin.cpp

```diff
@@ -172,7 +172,7 @@ void VstPlugin::stateBufferFromString(VstMemoryStream& buffer, char* strData, co
 
     static Steinberg::int32 numBytesRead = 0;
 
-    buffer.write(strData, strSize, &numBytesRead);
+    buffer.write(strData, strSize, &(static_cast<size_t>(numBytesRead)));

```

Comment:
This does not look right; the warning says that a `size_t` is being converted to `int32`. But here you are casting something that is already a `int32` to a `size_t`.

I think the warning is about the `strSize` parameter instead.

Reply:
oops, guess you're right ;-)